StRoot/StFwdTrackMaker/include/Tracker/TrackFitter.h

```diff
@@ -588,21 +604,84 @@ class TrackFitter {
             }
 
             TVector3 p = fitTrack.getCardinalRep()->getMom(fitTrack.getFittedState(1, fitTrack.getCardinalRep()));
-            // get status if needed later
-            // auto newStatus = fitTrack.getFitStatus(fitTrack.getCardinalRep());
+            return p;
+        }
+        return pOrig;
+    } // refit with Si hits
 
-            // try {
-            //     LOG_INFO << "Rechecking projected errors" << endm;
-            //     auto amsp2 = projectToFst(2, pFitTrack);
-            //     auto amsp1 = projectToFst(1, pFitTrack);
-            //     auto amsp0 = projectToFst(0, pFitTrack);
-            // }catch (genfit::Exception &e) {
-            // }
+    TVector3 refitTrackWithGBL( genfit::Track *originalTrack ) {
+        // mem leak, global track is overwritten without delete.
+        TVector3 pOrig = originalTrack->getCardinalRep()->getMom(originalTrack->getFittedState(1, originalTrack->getCardinalRep()));
+        
+        // auto cardinalStatus = originalTrack->getFitStatus(originalTrack->getCardinalRep());
 
-            return p;
+        if (originalTrack->getFitStatus(originalTrack->getCardinalRep())->isFitConverged() == false) {
+            // in this case the original track did not converge so we should not refit. 
+            // probably never get here due to previous checks
+            return pOrig;
+        }
+
+        // Setup the Track Reps
+        auto trackRepPos = new genfit::RKTrackRep(mPdgPositron);
+        auto trackRepNeg = new genfit::RKTrackRep(mPdgElectron);
+
+        // get the space points on the original track
+        auto trackPoints = originalTrack->getPointsWithMeasurement();
+        
+
+        TVectorD rawCoords = trackPoints[0]->getRawMeasurement()->getRawHitCoords();
+        TVector3 seedPos(rawCoords(0), rawCoords(1), rawCoords(2));
+        TVector3 seedMom = pOrig;
+
+        // Create the ref track using the seed state
+        auto pFitTrack = new genfit::Track(trackRepPos, seedPos, seedMom);
```

Comment:
Shouldn't this track be deleted later to avoid a memory leak?

Reply:
done in https://github.com/star-bnl/star-sw/pull/492/commits/3b3eaa9712529ed4b4922d1c48c694f26b74d38e